src/ansi-c/library/cprover_contracts.c

```diff
@@ -64,9 +64,24 @@ typedef struct
   void **elems;
 } __CPROVER_contracts_obj_set_t;
 
-/// \brief Type of pointers to \ref __CPROVER_contracts_car_set_t.
+/// \brief Type of pointers to \ref __CPROVER_contracts_obj_set_t.
 typedef __CPROVER_contracts_obj_set_t *__CPROVER_contracts_obj_set_ptr_t;
 
+/// \brief Stores context information supporting the evaluation of pointer
+/// predicates in both assume and assert contexts for all predicates:
+/// pointer equals, pointer_in_range_dfcc, pointer_is_fresh, obeys_contract.
+typedef struct
+{
+  /// \brief Nondet variable ranging over the set of objects allocated
+  /// by __CPROVER_contracts_is_fresh. Used to check separation constraints
+  /// in __CPROVER_contracts_is_fresh.
+  void *fresh_ptr;
+} __CPROVER_contracts_ptr_pred_ctx_t;
+
+/// \brief Type of pointers to \ref __CPROVER_contracts_ptr_pred_ctx_t.
+typedef __CPROVER_contracts_ptr_pred_ctx_t
+  *__CPROVER_contracts_ptr_pred_ctx_ptr_t;
+
```

Comment:
This new structure encapsulates context info that's used by pointer predicates and shared across requires and ensures clauses. It is going to be expanded in the future with more fields to support more predicates

Reply:
The mame is `ptr_pret_ctx` means  "pointer predicate context", it is not specific to is_fresh anymore (will be used by all predicates).